DSharpPlus/DiscordClient.cs

```diff
@@ -2338,7 +2336,7 @@ internal async Task OnHelloAsync(GatewayHello hello)
             }
             else
             {
-                this.DebugLogger.LogMessage(LogLevel.Warning, "DSharpPlus", "Session start attempt was made while another session is active", DateTime.Now);
+                this.Logger.LogWarning(ClientEventId, "Attempt to start a session while another session is active");
```

Comment:
My one question is why do warnings and Errors share the same Event ID as traces and informational messages.  Wouldn't a more granular approach help when parsing a huge log file so that we can hide certain event ids?  Granted yes the argument can be had that we shouldn't always be logging informational and trace messages but on the flip side, when we do have to log those messages, it would still help if we could filter that out at the beginning of an investigation and then once as we see an event id that we didn't filter out, we can then un-hide everything  surrounding that message.

Reply:
Event IDs are for the areas that are logged (think WebSocket, Rest, DSharpPlus, Interactivity, etc). If you want to filter out specific event ids I would assume you can pass in your own logger instance,

Reply:
As stated above, level != ID. If you want to filter warning messages from your logs, you can filter them out easily. ID is meant to state which component/area of the library emitted the event, and it's a diagnostic information. This creates the ability to filter by 2 different properties. I fail to see the point of separating each area into subareas depending on level.

Reply:
I totally agree @Emzi0767 that level != ID.  But however the numeric Id should not be shared between 2 different types of log messages.  My example was to show that the same numeric id is shared between 2 different log messages along that were also 2 different levels.  I will say that the numeric Id values shouldnt be all over the place (ex: 102, 503, 11055 for Rest Messages) and you should block off Numeric Id Values 100-199 for example would be for Rest Messages, Values 200-299 are reserved for DSharpPlus, Values 300-399 are for Interactivity, etc or however u would like to block it off.  That is what i was trying to originally state

Reply:
ID designates an event which emitted the message. It's not a message ID. The allocations are, in fact, done. For now, 1xx is core, 2xx is CNext, 3xx is VNext, 4xx is Lavalink, and 5xx is Interactivity IIRC.

Reply:
I encourage you to read [the documentation on the subject](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-3.1#log-event-id).

Reply:
@Emzi0767 i have read the documentation which is why i brought this up.  Also Notice how in that same documentation u sent, they use different numeric values for those 2 separate types of Log Messages.  One being MyLogEvents.GetItem which equaled 1002 and the other being a different numeric value for MyLogEvents.GetItemNotFound which equaled 4000.  However within your branch you have the EventId Struct defined here With an event Id of 100:
https://github.com/DSharpPlus/DSharpPlus/blob/2676da720fd915096aaa63cd2cacca65af51b9b9/DSharpPlus/BaseDiscordClient.cs#L20

And that same EventId is used here:
https://github.com/DSharpPlus/DSharpPlus/blob/2676da720fd915096aaa63cd2cacca65af51b9b9/DSharpPlus/DiscordClient.cs#L2339

and here: 
https://github.com/DSharpPlus/DSharpPlus/blob/2676da720fd915096aaa63cd2cacca65af51b9b9/DSharpPlus/DiscordClient.cs#L2458

and Here: 
https://github.com/DSharpPlus/DSharpPlus/blob/2676da720fd915096aaa63cd2cacca65af51b9b9/DSharpPlus/DiscordClient.cs#L990

This goes against what the Microsoft Documentation lays out in their example. Yes that documentation does say "An event ID associates a set of events. For example, all logs related to displaying a list of items on a page might be 1001."  but this is inferred as any informational messages not just all messages around this area hense why the example showed it as it did. Now, if you want to go against what it shows, okay that is fine and totally up to you but my initial question was to why.  Cause when someone says 1xx is for Chunk of code A, 2xx is for Chunk of code B, and 3xx is for Chunk of code C one expects more distinct Event Ids to be specified rather than just the 6 that are set and most of the log messages share the same Ids.

Reply:
Having thought this over a little, I'm inclined to agree, to some extent.